gc/base/standard/Scavenger.cpp

```diff
@@ -5932,6 +5943,33 @@ MM_Scavenger::completeConcurrentCycle(MM_EnvironmentBase *env, MM_MemorySubSpace
 	}
 }
 
+void
+MM_Scavenger::payAllocationTax(MM_EnvironmentBase *env, MM_MemorySubSpace *subspace, MM_MemorySubSpace *baseSubSpace, MM_AllocateDescription *allocDescription)
+{
+	if (isCurrentPhaseConcurrent()) {
+		uintptr_t freeMemory = _activeSubSpace->getMemorySubSpaceAllocate()->getApproximateActiveFreeMemorySize();
+		uintptr_t totalMemory = _activeSubSpace->getMemorySubSpaceAllocate()->getActiveMemorySize();
+
+		/* Progress of how much Survivor/Allocate area has been consumed since the start this cycle relative.
+		 * (Although fairly small, consider excluding the amount consumed by the first STW increment.
+		 */
+		float usedMemoryRatio = 1.0f - (float)freeMemory / totalMemory;
+
+		/* Total flipped bytes, from both GC background threads and from Read Barriers, since the start of the concurrent phase of this cycle. */
+		uintptr_t flipBytes = _extensions->incrementScavengerStats._flipBytes + _extensions->incrementScavengerStats._readObjectBarrierFlipBytes;
+
+		/* Progress of how much objects have been flipped since the start of concurrent phase of this cycle relative to historically averaged amount. */
+		float flipBytesRatio = (float)flipBytes / _extensions->scavengerStats._avgExpectedFlipBytes;
+
+		/* If flip progress is too low (too small Nursery, background GC threads starved by CPU overloading,...)
+		 * mutator threads need to be taxed to slow down their heap allocation and CPU consumption. */
+		if ((1.0f + flipBytesRatio < usedMemoryRatio)) {
+			/* Yet to provide a meaningful heuristic (current one should never trigger) */
```

Comment:
added period, consistency
```suggestion
			/* Yet to provide a meaningful heuristic (current one should never trigger). */
```